Description: add missing main.moc include for plasmarunner and use cmake settings
 why exactly the build works for anyone without the moc is beyond me, the
 include however is supposedly necessary as documented at [1]. otherwise
 the factory (object created by the macro) metaobject code is not present
 and the factory linking fails on the missing vtable. this is reproducible
 on cmake 3.2.2, gcc5 and frameworks 5.13.
 .
 additionally include the KDECMakeSettings from ECM as to enable
 current_binary_directory includes by default (which is necessary to
 find main.moc)
 .
 [1] https://techbase.kde.org/Development/Tutorials/Plasma4/AbstractRunner#The_Plugin_Declaration_Macro
Author: Harald Sitter <sitter@kde.org>
Forwarded: https://git.reviewboard.kde.org/r/124787/
Origin: vendor
diff --git a/src/plasmarunner/CMakeLists.txt b/src/plasmarunner/CMakeLists.txt
index 32591ef6bfc7b41ca644f5ec34fbdb97225f5749..7a68d097cb36a86dd83d66c43f0cd32d89a3476d 100644
--- a/src/plasmarunner/CMakeLists.txt
+++ b/src/plasmarunner/CMakeLists.txt
@@ -5,6 +5,7 @@ find_package(ECM QUIET)
 set(CMAKE_MODULE_PATH ${CMAKE_MODULE_PATH} ${ECM_MODULE_PATH})
 if(ECM_FOUND)
   include(KDEInstallDirs)
+  include(KDECMakeSettings)
 endif()
 include(FeatureSummary)
 macro_optional_find_package(KF5 QUIET COMPONENTS Config Runner Service I18n)
@@ -12,7 +13,7 @@ macro_optional_find_package(KF5 QUIET COMPONENTS Config Runner Service I18n)
 if (NOT KF5_FOUND)       
   return()
 endif()
-
+  
 set(plasmaRunner_SRCS
   main.cpp
   plasmarunner.cpp
diff --git a/src/plasmarunner/main.cpp b/src/plasmarunner/main.cpp
index 43c30bb2dcb3d078e6f214aaab23e2688399354f..3f60f9fab149ffc6a862f03af770b077b991e894 100644
--- a/src/plasmarunner/main.cpp
+++ b/src/plasmarunner/main.cpp
@@ -17,3 +17,4 @@
 
 K_EXPORT_PLASMA_RUNNER(marble, Marble::PlasmaRunner)
 
+#include "main.moc"
